src/GameLogic/PlugIns/InvasionEvents/HappyHourPlugIn.cs

```diff
@@ -0,0 +1,220 @@
+﻿// <copyright file="HappyHourPlugIn.cs" company="MUnique">
+// Licensed under the MIT License. See LICENSE file in the project root for full license information.
+// </copyright>
+
+namespace MUnique.OpenMU.GameLogic.PlugIns.InvasionEvents;
+
+using System.Collections.Concurrent;
+using System.Runtime.InteropServices;
+using MUnique.OpenMU.GameLogic.NPC;
+using MUnique.OpenMU.GameLogic.Views;
+using MUnique.OpenMU.PlugIns;
+
+public class HappyHourConfiguration : PeriodicInvasionConfiguration
```

Comment:
Will refactor it, because we need base class like `PeriodicEventConfiguration`

Another part - GameServerState